### PR TITLE
Ilari's proposal

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-ietf-cose-hpke.md [hunk 1]
```diff
@@ -47,6 +47,9 @@ key derivation function (KDF), and authenticated encryption with
 additional data (AEAD) encryption function. Authentication for HPKE in COSE is
 provided by COSE-native security mechanisms.
 
+This document defines the use of the HPKE base mode with COSE. Other modes are
+supported by HPKE but those are not supported by this specification.
```
Comment (hannestschofenig) [inline draft-ietf-cose-hpke.md:51]:
```suggestion
supported by HPKE but not by this specification.
```

File: draft-ietf-cose-hpke.md [hunk 2]
```diff
@@ -94,69 +97,68 @@ This specification supports two uses of HPKE in COSE, namely
...
+   +---------+----------------+------------+-------------------+
 ~~~
-{: #table-hpke-sender title="COSE_HPKE_Sender Labels"}
+{: #table-hpke-sender title="encapsulated_key structure"}
 
-   kem id:  This parameter is used to identify the Key Encapsulation
+  kem_id: This parameter is used to identify the Key Encapsulation
        Mechanisms (KEM). The registry for KEMs has been established
        with RFC 9180. This parameter is optional since the kid may be
        used to determine the KEM.
```
Comment (laurencelundblade) [inline draft-ietf-cose-hpke.md:150]: Text says it is optional. It isn't anymore, right?

File: draft-ietf-cose-hpke.md [hunk 3]
```diff
@@ -94,69 +97,68 @@ This specification supports two uses of HPKE in COSE, namely
...
+   |         |                | Registry   | Encapsulation     |
+   |         |                |            | Mechanisms        |
+   |         |                |            |                   |
+   | kdf_id  | uint           | HPKE KDF   | Identifiers for   |
+   |         |                | IDs        | KDF IDs           |
+   |         |                |            |                   |
+   | aead_id | uint           | HPKE AEAD  | Identifiers for   |
+   |         |                | IDs        | AEAD IDs          |
+   |         |                |            |                   |
+   | enc     | bstr           |            | Encapsulated key  |
```
Comment (laurencelundblade) [inline draft-ietf-cose-hpke.md:141]: If the whole array is the "encapsulated key" then this array element shouldn't also have the same name.

**Review by laurencelundblade — COMMENTED** [review]
Looks good. Just a few wording suggestions.

File: draft-ietf-cose-hpke.md (hunk 3 above)
Comment (dajiaji) [reply draft-ietf-cose-hpke.md:141]: > If the whole array is the "encapsulated key" then this array element shouldn't also have the same name.

I believe that "encapsulated key" was originally named to accommodate not only HPKE encapsulated keys, but also other encapsulated keys. However, currently this structure is dedicated to HPKE and kdf_id and aead_id are not related to "encapsulated key".

I understand it's Bikeshed, but I think the name of the whole array should be "HPKE sender information (for base mode)" as defined in #9.

**Pull request merged**